src/drivers/uavcan/sensors/gnss.cpp

```diff
@@ -463,9 +480,17 @@ void UavcanGnssBridge::process_fixx(const uavcan::ReceivedDataStructure<FixType>
 		report.vdop = msg.pdop;
 	}
 
-	report.heading = heading;
-	report.heading_offset = heading_offset;
-	report.heading_accuracy = heading_accuracy;
+	// Use heading from RelPosHeading message if available and we have RTK Fixed solution.
+	if (_rel_heading_valid && (fix_type == sensor_gps_s::FIX_TYPE_RTK_FIXED)) {
```

Comment:
I wonder if we should also have a timestamp associated with the rel_heading. I'm not sure if this can happen, but I'm imagining a case where we have heading from moving base, but then it goes away and comes back as an RTK fix but without the heading...